lib/mobile_provision/representation.rb

```diff
@@ -93,5 +98,19 @@ def build_registered_udids
       return nil unless @profile_type == AD_HOC_TYPE
       read_plist_value(ADHOC_PROFILE_KEY)
     end
+
+    def build_team_id
+      entitlements = read_plist_value(ENTITLEMENTS_KEY)
+      read_plist_value(TEAM_ID_KEY, entitlements)
+    end
+
+    def build_bundle_id
+      build_app_id[/(?<=\.).*/]
+    end
+
+    def has_associated_domains?
+      entitlements = read_plist_value(ENTITLEMENTS_KEY)
+      read_plist_value(HAS_ASSOCIATED_DOMAINS, entitlements).kind_of? String
+    end
```

Comment:
In my opinion this method should be an alias of the accessor/reader:

```ruby
def has_associated_domain
  # ...
end
alias_method :has_associated_domain?, :has_associated_domain
```

Also, the return value should probably memoized so that the `read_plist_value` method is not called every single time:

```ruby
def has_associated_domain
  return @has_associated_domain if defined?(@has_associated_domain)
  entitlements = read_plist_value(ENTITLEMENTS_KEY)
  @has_associated_domain = read_plist_value(HAS_ASSOCIATED_DOMAINS, entitlements).kind_of?(String)
end
alias_method :has_associated_domain?, :has_associated_domain
```